bin/helm-generate-raw-manifests-in-docker

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -eo pipefail
+
+diff_only=false
+
+while true; do
+  case "$1" in
+    -d|--diff-only )
+      diff_only=true
+      shift
+      ;;
+    * )
+      if [ -z "$1" ]; then
+        break
+      else
+        echo "$1 is not a valid option"
+        exit 1
+      fi
+      ;;
+  esac
+done
+
+echo "Building utility image with helm and yq..."
+docker build -f Dockerfile.helm-unit-test -t conjur-k8s-helm-generate-manifests:dev .
+
+echo "Generating raw k8s manifests..."
+docker run --name conjur-k8s-helm-generate-manifests --entrypoint "/conjur-authn-k8s-client/bin/helm-generate-raw-manifests" \
+            conjur-k8s-helm-generate-manifests:dev
+
+set +eo pipefail
+
+docker cp conjur-k8s-helm-generate-manifests:/conjur-authn-k8s-client/helm/conjur-config-cluster-prep/generated/conjur-config-cluster-prep.yaml \
+            helm/conjur-config-cluster-prep/generated/conjur-config-cluster-prep.new.yaml
+
+docker cp conjur-k8s-helm-generate-manifests:/conjur-authn-k8s-client/helm/conjur-config-namespace-prep/generated/conjur-config-namespace-prep.yaml \
+            helm/conjur-config-namespace-prep/generated/conjur-config-namespace-prep.new.yaml
+
+docker rm conjur-k8s-helm-generate-manifests
+
+if [[ "$diff_only" = true ]];
+then
+    set -eo pipefail
+    echo -e "\n*** If generated files have changed please run \"bin/helm-generate-raw-manifests-in-docker\" and commit the new manifests ***"
+fi
+
+echo -e "\nDiffing generated/conjur-config-cluster-prep.yaml..."
+diff -cs helm/conjur-config-cluster-prep/generated/conjur-config-cluster-prep.yaml \
+            helm/conjur-config-cluster-prep/generated/conjur-config-cluster-prep.new.yaml
+
+echo -e "\nDiffing generated/conjur-config-namespace-prep.yaml..."
+diff -cs helm/conjur-config-namespace-prep/generated/conjur-config-namespace-prep.yaml \
+            helm/conjur-config-namespace-prep/generated/conjur-config-namespace-prep.new.yaml
+
+if [[ "$diff_only" = false ]];
+then
+    echo -e "\nOverwriting old manifests..."
```

Comment:
If `$diff_only` is set to false, I think this will always echo the "Overwriting old manifests", even if they haven't changed. It might be nicer to check the results of the `diff -cs ...` commands above, and then do the overwrite only if the files have actually changed.

Another thought... If you do add checks for the results of the `diff -cs ...`, you should be able to remove the `If generated files have changed...` display above, and instead echo something more definitive here when the files have actually changed, e.g. `Generated files have changed. Please run ....`.

Reply:
Changed